R/conditions.R

```diff
@@ -51,7 +51,7 @@ with_handlers <- function(code, handlers) {
 }
 
 sanitize_call <- function(cnd) {
-  if (identical(cnd$call, quote(eval(expr, envir)))) {
+  if (identical(cnd$call, quote(withVisible(do)))) {
```

Comment:
Is this the sanitation code that should trigger for knitr for the error ? 

I have a hard time for now understanding what `do` will be here. When is the identical supposed to be `TRUE` ? 

I am under the impression that this is not the expected value when ran inside **knitr** evaluation call, and why the sanitation may not happen.  When doing `knitr::knit()`, we definitely go into this function. 

Also not using **knitr** is throwing the same error not sanitized 
````
> evaluate::evaluate("mpg")
<evaluation>
Source code: 
  mpg
Condition: 
  Error in eval(as.call(list(cb)), envir):
  objet 'mpg' introuvable
````

I feel I am missing something here, or this is currently not sanitizing as we would like

Reply:
Yeah, definitely not sanitising. The weird thing is that I made the change here in order to get the unit tests passing, so it's clearly working in some circumstances.

Reply:
Ah the problem is that this sort of error seems to get a different call attached to it — maybe because the error occurs at the very top-level, not from a call to `stop()`

Reply:
This is indeed working as expected in **knitr** now. 

> that would also explain why knitr included the call, even though it also tries to strip the call.

AFAIK knitr will strip sometimes for warning, but for error, it seems to me it won't strip it. One need to opt-in with knitr sanitation feature through option `knitr.sanitize.errors = TRUE` and it works

Reply:
Ah that makes sense. It's a bit inconsistent that it always sanitises warnings but only sanitises errors if you ask.